src/tracks/playabletrack/wavetrack/ui/WaveTrackControls.cpp

```diff
@@ -741,56 +743,99 @@ void WaveTrackMenuTable::OnMergeStereo(wxCommandEvent &)
    AudacityProject *const project = &mpData->project;
    auto &tracks = TrackList::Get( *project );
 
-   WaveTrack *const pTrack = static_cast<WaveTrack*>(mpData->pTrack);
-   wxASSERT(pTrack);
+   const auto first = tracks.Any<WaveTrack>().find(mpData->pTrack);
+   const auto left = *first;
+   const auto right = *std::next(first);
 
-   auto partner =
-      static_cast<WaveTrack*>(*tracks.Find(pTrack).advance(1));
-
-   if (pTrack->GetRate() != partner->GetRate()) {
-      using namespace BasicUI;
-      ShowMessageBox(XO(
-"Mono tracks must have the same sample rate in order to be combined into a "
-"stereo track"),
-         MessageBoxOptions{}
-           .Caption(XO("Error"))
-           .IconStyle(Icon::Error));
-      return;
-   }
-   if(pTrack->GetSampleFormat() != partner->GetSampleFormat())
+   const auto checkAligned = [](const WaveTrack& left, const WaveTrack& right)
    {
-      BasicUI::ShowMessageBox(XO(
-"Mono tracks must have the same sample format in order to be combined "
-"into a stereo track"),
```

Comment:
Why are the tests for sameness of rate and format removed?

Reply:
I think it was implied that these checks are not needed any more
![изображение](https://github.com/audacity/audacity/assets/20165230/e6c88220-3313-498d-8870-5b2641155d8e)

Reply:
We're doing a destructive mix and render anyway which resamples on the way anyway, does it not?

Reply:
Yes.

The past work to disallow mismatched rates or formats was not wasted because other deduplicarion in the data structures was the important accomplishment in those PRs.